src/openforms/js/components/admin/submissions/filter.js

```diff
@@ -0,0 +1,47 @@
+const init = async () => {
+  // Retrieve the filter link that displays only triggered rules
+  const triggeredLink = document.querySelector('.logic-rules-filter > .selected > a');
+
+  // Only triggered rules are displayed
+  document
+    .querySelectorAll('.logic-logs__col-body--trigger-' + triggeredLink.dataset.trigger)
+    .forEach(trigger => {
+      trigger.parentNode.classList.remove('logic-logs__row-body--hidden');
+    });
+};
+
+document.querySelectorAll('.logic-rules-filter__link').forEach(link => {
+  link.addEventListener('click', event => {
+    event.preventDefault();
+
+    // Hiding all the rows first
+    document.querySelectorAll('.logic-logs__row-body').forEach(row => {
+      row.classList.add('logic-logs__row-body--hidden');
+    });
+
+    const filterValue = event.target.dataset.trigger;
+
+    // Change the focus of the selected filter
+    document.querySelector('.selected').classList.remove('selected');
+    event.target.parentNode.classList.add('selected');
+
+    // Showing only rules corresponding to the selected trigger link
+    if (filterValue == 'all') {
+      document.querySelectorAll('.logic-logs__row-body').forEach(row => {
+        row.classList.remove('logic-logs__row-body--hidden');
+      });
+    } else {
+      document
+        .querySelectorAll('.logic-logs__col-body--trigger-' + filterValue)
+        .forEach(trigger => {
+          trigger.parentNode.classList.remove('logic-logs__row-body--hidden');
+        });
+    }
```

Comment:
you can clean up this code to avoid the repetition:

```suggestion
    // Showing only rules corresponding to the selected trigger link
    let rows;
    switch (filterValue) {
      case 'all': {
        rows = document.querySelectorAll('.logic-logs__row-body');
        break;
      }
      case 'yes': {
        rows = document.querySelectorAll('.logic-logs__row-body.logic-logs__col-body--trigger-yes');
        break;
      }
      case 'no': {
        rows = document.querySelectorAll('.logic-logs__row-body.logic-logs__col-body--trigger-no');
        break;
      }
    }
    for (const row of rows) {
      row.classList.remove('logic-logs__row-body--hidden');
    }
```